fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AlterSystemStmtAnalyzer.java

```diff
@@ -50,5 +57,35 @@ public Void visitFrontendClause(FrontendClause frontendClause, ConnectContext co
             }
             return null;
         }
+
+        @Override
+        public Void visitModifyFrontendHostClause(ModifyFrontendAddressClause clause, ConnectContext context) {
+            try {
+                String fqdn = clause.getFqdn();
+                // if fqdn is a domain name need to determine whether it is a legitimate domain name
+                if (NetUtils.validIPAddress(fqdn)) {
+                    throw new SemanticException("the host you want to set could't be an ip");
+                }
+                InetAddress.getByName(fqdn);
+            } catch (UnknownHostException e) {
+                throw new SemanticException("unknown host " + e.getMessage());
+            }
+            return null;
+        }
+
+        @Override
+        public Void visitModifyBackendHostClause(ModifyBackendAddressClause clause, ConnectContext context) {
+            try {
+                String fqdn = clause.getFqdn();
+                // if fqdn is a domain name need to determine whether it is a legitimate domain name
+                if (NetUtils.validIPAddress(fqdn)) {
```

Comment:
If you should reject the operation when the srcHost and the destHost are the same, it's non-sense.